clusterloader2/testing/l4ilb/config-ilb-recovery.yaml

```diff
@@ -0,0 +1,124 @@
+# Timer measurement is used to measure the time between the start of an action executed by the
+# Exec command and when the ILB services are running and reachable.
+# ServiceCreationLatency measurement is used only for determining the moment when all of the
+# ILBs are reachable. Since it reports time based on the initial creation of the service, and
+# not from the start of the action, the measured time will not be shown.
```

Comment:
What does this mean? We will have a timeout of 10m(by default) for ILBs to be ready after starting the action, but we will not export the total time taken?

Reply:
The Timer measurement is used here for getting the time that is required for services to be reachable after an action.
The ServiceCreationLatency measurement always measures and reports time based on the initial creation time of the service, and not when the action is triggered. Since recreating the nodes will only reschedule the services, the ServiceCreationLatency would use the initial creation time of the ILBs as the start time, regardless of when we start the measurement.
That’s why the time measured by it is not useful for this test, but we can use it anyway only for getting the information when the services are all reachable again.

Reply:
Your statements are correct.
If we only want to measure the time from the moment when nodepool upgrade ends, we can just move the Timer to start after the EXEC command is finished.
We can also add another Timer to measure multiple periods.

What do you suggest?

Reply:
Ideally, we want the timer to measure the time for which ILB services were unreachable/flaky, right from when nodepool upgrade command got kicked off. Your change is already doing that. I think the issue is - even if there was no disruption, the test result will show the time interval as non-zero (time taken for nodepool upgrade to finish). So it won't really be useful.

A more useful measure is - count the number of requests to ILB VIP that resulted in failure. (non-200 status code). The timer value is fine to measure the nodepool upgrade duration. The failed_count will give us an idea of disruption and measure any performance improvement we make.

Reply:
To measure success rate of requests to ILB during the nodepool upgrade requires a new measurement.
This will be next on the list for required measurements for load balancers. It will be added to this test once it is implemented.